obspy/core/inventory/util.py

```diff
@@ -859,6 +862,139 @@ def _seed_id_keyfunction(x):
     return x
 
 
+def plot_inventory_epochs(plot_dict, outfile=None):
+    """
+    Creates a plot from inventory object's epoch plottable structure.
+    :param plot_dict: Dictionary of inventory epochs. A structure used and
+        created by the given inventory/station/network/channel object that
+        contains the epoch data for an inventory in a tuple with the format
+        (start_time, end_time, sub_dict) where sub_dict is a dictionary with
+        the same type of structure for any subcomponents.
+    :type plot_dict: dict
+    :param outfile: If this parameter is included, the plot will be saved to
+        a file with the given filename.
+    :type outfile: str
+    """
+
+    # dictionary will hold plot component's initial y-value & height
+    y_dict = {}
+    # need to do tree traversal for getting appropriate y-axis variables
+    _plot_traversal_helper_(plot_dict, y_dict)
+
+    # get height of each inv. component data, color lines according to height
+    y_tick_labels = []
+    y_ticks = []
+    y_min = float('inf')
+    y_max = 0
+    clr_dict = {}
+    for key in y_dict.keys():
+        (tick, height) = y_dict[key]
+        if height == 1:
+            y_tick_labels.append(key)
+            y_ticks.append(tick)
+            y_min = min(tick-1, y_min)
+            y_max = max(tick+height, y_max)
+    clrs = iter(cm.Dark2(linspace(0, 1, len(y_tick_labels))))
+    for label in sorted(y_tick_labels):
+        clr_dict[label] = next(clrs)
+
+    plt.figure()
+    ax = plt.gca()
+    # get the plot ranges
+    xmax = 0
+    xmin = float(UTCDateTime.now())
+    # add the plottable data to the plot
+    (xmin, xmax) = _plot_builder_(ax, plot_dict, y_dict, xmin, xmax, clr_dict)
+    xmax = min(xmax, float(UTCDateTime.now()))
+    ax.set_xlim(xmin, xmax)
+    ax.set_ylim(y_min, y_max)
+
+    # label the x-axis according to given time data
+    x_ticks = ax.get_xticks()
+    x_tick_labels = []
+    for tick in x_ticks:
+        label = str(UTCDateTime(tick).date)
+        x_tick_labels.append(label)
+    plt.yticks(y_ticks, y_tick_labels)
+    plt.xticks(x_ticks, x_tick_labels, rotation='45')
+    plt.grid()
+
+    if outfile:
+        plt.savefig(outfile)
+    else:
+        plt.show()
+
+
+def _plot_traversal_helper_(plot_dict, y_dict, offset=0, prefix=''):
```

Comment:
no trailing underscore with method/function names, please, to be in line with rest of code base

Reply:
I was doing that as a means of marking the functions that weren't intended to be callable by the user. Is the leading underscore still acceptable, or should I remove that as well?

Reply:
A single leading underscore is correct Python practice for private routines, just get rid of the trailing one please.

Reply:
Leading underscore denotes "private" function. Leave it.